### ISSUE TITLE
HF epsilon

### ISSUE THREAD

**Issue by SunByrne** [singmann/afex#36]
When the HF epsilon is high and HF is set as the correction type, aov_ez fails with this error message:

```
Error in if (any(tmp[["pval.adjustments"]][, "HF eps"] > 1)) warning("HF eps > 1 treated as 1",  : 
  missing value where TRUE/FALSE needed
```

**Comment by singmann** [comment]
Can you provide a reproducible example? It seems to work for me, see:

```
require(afex)

data(md_12.1)
aov_ez("id", "rt", md_12.1, within = c("angle", "noise"), 
       anova_table=list(correction = "HF", es = "none"))
# Anova Table (Type 3 tests)
# 
# Response: rt
#        Effect    df     MSE         F p.value
# 1       angle 2, 18 3560.00 40.72 ***  <.0001
# 2       noise  1, 9 8460.00 33.77 ***   .0003
# 3 angle:noise 2, 18 1160.00 45.31 ***  <.0001
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘+’ 0.1 ‘ ’ 1
# 
# Sphericity correction method: HF 
# Warning message:
# HF eps > 1 treated as 1 
```

**Comment by singmann** [comment]
Did your problem solve itself or can you provide a reproducible example?

**Comment by SunByrne** [comment]
No, it is not resolved, happens every time with a particular data set, but I can't release those data until I've had a chance to discuss it with my collaborators. Unfortunately, I live in the Houston area and have had other things to deal with of late.

**Comment by singmann** [comment]
I am sorry to hear that. 
Unfortunately, if I cannot reproduce the error I cannot really track it down or provide a patch. If you want, you can send me the data per mail and I will treat it confidentially. Otherwise, I will just wait until you get back to me.

**Comment by SunByrne** [comment]
OK, I know it's been a while, but now I've gotten the same error with a totally different data set, and this one is easy to share.

`demo = read.delim("http://byrneweb.com/sunburn/tmp/demo.txt")

afex_options(correction_aov="none")
aov_ez("Snum", "RT", demo, within=c("DistF", "WidthF", "AngleF"))


afex_options(correction_aov="HF")
aov_ez("Snum", "RT", demo, within=c("DistF", "WidthF", "AngleF"))`

Here's the output of the first:

`Anova Table (Type 3 tests)

Response: RT
               Effect       df      MSE          F  ges p.value
1               DistF    2, 52 20782.52 415.77 ***  .18  <.0001
2              WidthF    3, 78 23913.05  15.84 ***  .01  <.0001
3              AngleF   7, 182 21079.14       1.11 .002     .36
4        DistF:WidthF   6, 156 18737.25       1.14 .002     .34
5        DistF:AngleF  14, 364 19007.23       0.81 .003     .66
6       WidthF:AngleF  21, 546 19249.29       1.19 .006     .25
7 DistF:WidthF:AngleF 42, 1092 17883.53    1.61 **  .01    .008---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘+’ 0.1 ‘ ’ 1
>`

And here's what I get with the second:

`Error in if (any(tmp[["pval.adjustments"]][, "HF eps"] > 1)) warning("HF eps > 1 treated as 1",  : 
  missing value where TRUE/FALSE needed
> `

Just the error message, no ANOVA table.

**Comment by singmann** [comment]
fixed in 7591da47893b78eb67843c2dff0c4bc1e6c8b64b.
Will submit the new package version to CRAN asap (probably today).
